pkg/acs/calacs/lib/mkspt.c

```diff
@@ -191,7 +229,50 @@ int mkNewSpt (char *in_list, char *mtype, char *output) {
         freeHdr (&header);
     }
 
+    /* If necessary, re-open the primary header to update the NEXTEND keyword */
+    if ((extnum != numSPT) || (numSPT != nimgs)) {
+
+        imUpdate = openUpdateImage (out_spt, "", 0, &header);
+        if (hstio_err()) {
+            trlopenerr (out_spt);
+            freeHdr (&header);
+            return (status = OPEN_FAILED);
+        }
+
+        if (PutKeyInt (&header, "NEXTEND", extnum, "")) {
+            reportErrorAndFreeHeader (&header);
+            closeImage (imUpdate);
+            return (status = hstio_err());
```

Comment:
Why return `hstio_err()`? This may not be correct, e.g. are you sure that the caller is expecting an enumeration in this range, of this type ([HSTIOERROR_](https://github.com/spacetelescope/hstcal/blob/master/include/hstio.h#L387-L393)) rather than those from [hstcalerr.h](https://github.com/spacetelescope/hstcal/blob/master/include/hstcalerr.h)? Furthermore, I checked a couple of the calls and they just pass `status` back up, so it's not just the immediate callers that this affects.
 
Furthermore, this shouldn't be bypassing the error handling by the called func.

Reply:
I double-checked this today to verify my original rationale. If one drills down through the PutKeyXXX functions, the lowest level code invokes the HSTIO error() function which sets an error_status and an error_msg.  This new code prints the fundamental reason for the error to the trailer file, and returns (as you note) the HSTIO error value. All the intermediate functions between PutKeyXXX and the low-level functions return 0, -1, or NULL.  Some of the intermediate functions also call HSTIO error().  At the level just below PutKeyXXX, the HSTIO putKeyXXX returns 0/1 based upon hstio_err().  I felt this information was more informative for the user than the current situation.  I can add the error code, for what it is worth, to the reportErrorAndFreeHeader function.

All existing functionality invoking mkNewSpt is just checking for 0/non-zero.  Having said this, I agree that I perhaps returning hstio_err() may be questionable in a general sense.  More tomorrow.

Reply:
Ultimately, I agree with you so status is now saved from the PutKeyXXX/putXXX functions and returned to the calling function on exit.  Before exiting, status is passed to a helper function with checks for specific errors in order to provide informational messages to the user.

Reply:
In order to avoid duplicate output trailer messages, the status of the PutKeyXXX functions is obtained and passed back to the calling function on error.  The "helper" function is not necessary any longer and was deleted.